suse_migration_services/suse_product.py

```diff
@@ -117,3 +117,24 @@ def get_tag(self, tag):
                 'Parsing XML file {0} failed with: {1}'
                 .format(self.base_product, issue)
             )
+
+    def get_product_name(self):
+        """Get the product name to be migrated to."""
+        migration_product_name = None
+        try:
+            name = self.get_tag('name')[0]
+            arch = self.get_tag('arch')[0]
+            if name and arch:
+                migration_product_name = '/'.join(
+                    [name, str(self.get_default_target_version()), arch]
+                )
+        except Exception as issue:
+            log.error(
+                'No auto detection of baseproduct: {0}. {1}'
+                .format(issue, 'Continue without auto detected product')
+            )
+
+        return migration_product_name
+
+    def get_default_target_version(self):
+        return 15
```

Comment:
Ok here we have a problem. Thinking further and assuming there is SLES-16 available. We design an image that upgrades from sles15 to sles16. The live iso image that drives the process will be a sles16 system and the user installs the package that provides the live iso image on his sles15 system. This code will now construct the product name ```SOME_NAME/15/SOME_ARCH``` and it's for sure wrong.

I suggest that the target version is detected from the ```/etc/os-release``` file VERSION_ID variable taken from the live iso images, ,meaning just read ```/etc/os-release```

On the currently designed image this will be 15 on a potential later image this will be 16 and so on

If we add auto detection then real auto detection please :)

Thoughts ?

Reply:
In addition I should also mention that the people who write the zypper migration plugin explicitly stated that it only works if the plugin is already called from the target system. This chicken & egg problem is solved by us with running it from the live iso which is always build for that target. So reading the os-release file is also the correct detection way and in line with the design concept

Reply:
Fixed.